src/components/SearchBar/SearchBar.hook.ts

```diff
@@ -0,0 +1,77 @@
+import { useState, useEffect, type ChangeEvent, type KeyboardEvent } from 'react';
+
+import { DEBOUNCE_DELAY } from './SearchBar.constants';
+
+import type { SearchBarProps } from './SearchBar.types';
+
+export const useSearchBar = (props: SearchBarProps) => {
+  const {
+    value: controlledValue,
+    defaultValue,
+    onSearch,
+    debounceDelay = DEBOUNCE_DELAY,
+    onChange,
+  } = props;
+
+  const isControlled = controlledValue !== undefined;
+  const [internalValue, setInternalValue] = useState(defaultValue || '');
+
+  // Sync internal state with controlled value
+  useEffect(() => {
+    if (isControlled) {
+      setInternalValue(controlledValue);
+    }
+  }, [controlledValue, isControlled]);
+
+  // Debounce logic
+  useEffect(() => {
```

Comment:
The debounce effect is present, but the actual onSearch call is commented out. This makes debounceDelay a no-op and is inconsistent with the component docs/stories that advertise debounced input handling. Decide whether the component should auto-trigger onSearch while typing (debounced) or only trigger on Enter/button and update code/docs/tests accordingly.

**Sugerencia:**
```typescript
If you want auto-search on typing, call onSearch with a safe value (see typing fix) inside the debounced handler. If you don't want auto-search, remove debounceDelay and this effect to avoid dead code and surprising API. Alternatively expose an explicit prop (e.g. autoSearch?: boolean) to control behavior.
```